### PR TITLE
Rjava v0.9 11

### PR DESCRIPTION
**Pull request by agcopenhaver**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by agcopenhaver** [comment]
@conda-forge-admin, please rerender

**Comment by agcopenhaver** [comment]
@conda-forge/r @sodre is this good to go?

**Comment by jdblischak** [comment]
@agcopenhaver Thanks for working on this!

l'm not qualified to review changes to a build script that involves `R CMD javareconf`. I worry how this will interact with [similar calls in the r-base recipe](https://github.com/conda-forge/r-base-feedstock/search?q=javareconf&unscoped_q=javareconf).

Tagging some folks that would be able to provide a review: @mbargull @isuruf @mingwandroid

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,14 +26,15 @@ requirements:
     - posix                # [win]
     - {{ compiler('c') }}        # [unix]
     - {{ compiler('m2w64_c') }}  # [win]
+
   host:
     - r-base
     - openjdk 8.*
-
+
```
Comment (sodre) [inline recipe/meta.yaml]:
```suggestion

```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,14 +26,15 @@ requirements:
...
     - r-base
     - openjdk 8.*
-
+    
   run:
     - r-base
-    - openjdk 8.*
-
+    - openjdk 8.*  
+
```
Comment (sodre) [inline recipe/meta.yaml]:
```suggestion

```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -26,14 +26,15 @@ requirements:
...
   host:
     - r-base
     - openjdk 8.*
-
+    
   run:
     - r-base
-    - openjdk 8.*
-
+    - openjdk 8.*
```
Comment (sodre) [inline recipe/meta.yaml]:
```suggestion
    - openjdk 8.*
```

**Review by sodre — CHANGES_REQUESTED** [review]
Minimize white-space changes...

**Comment by dbast** [comment]
Both r-base scripts contain `R CMD javareconf`:
* https://github.com/conda-forge/r-base-feedstock/blob/master/recipe/activate-r-base.sh
* https://github.com/conda-forge/r-base-feedstock/blob/master/recipe/post-link.sh

Why this is not enough and why adding `R CMD javareconf` also to the build.sh here is fixing the osx build, looks a bit mysterious. But it should be of no harm to have multiple calls of javareconf during build.

File: recipe/meta.yaml (hunk 3 above)
Comment (agcopenhaver) [reply recipe/meta.yaml]: my bad. Don't know why I did that

**Comment by mbargull** [comment]
My best guess is that this is due to `conda-build`'s environment stacking:
1. (`host` environment's `r-base` post-install script runs `javareconf`)
2. `host` environment gets activated / `host` environment's activation scripts are run \*\*
   1. compiler activation scripts set `CC`, `CFLAGS`, ... with path to relative to `$PREFIX`
         (the compilers are installed into `host` as a dependency of `r-base`)
   2. `r-base` activation script runs `javareconf`
3. `build` environment gets activated / `build` environment's activation scripts are run
   - compiler activation scripts set `CC`, `CFLAGS`, ... with path to relative to `$BUILD_PREFIX`

(\*\* NOTE: Unless this has been fixed in `conda` (didn't check), sadly, the activation scripts are not guaranteed to run in dependency order. AFAIR, they are run in lexicographical order of their filenames. In this case, `activate-{binutils,gcc,...}*.sh` would lexicographically precede `activate-r-base.sh`, so we should get the intended order, thankfully.)

If `javareconf` is influenced by the compiler's environment variables or other path-dependent compiler output, then this would be a likely explanation since `javareconf` is only run in the `host` post-install or activation before `build` gets activated.
=> Running `javareconf` in the build script is likely the right call when there are stacked `host`+`build` environments.

@mingwandroid, @conda-forge/r : I assume this might affect all R packages that compile Java-related things and depend on Java (directly or indirectly through `r-rjava`). Meaning, those packages should likely run `"${PREFIX}/bin/R" CMD javareconf` in their `build.sh`.

NB: I don't use R and macOS. Hence I don't know the peculiarities of R and can't test locally. And I have no idea why only macOS but not Linux failed.

**Review by mbargull — DISMISSED** [review]
LGTM, I guess (...says the not fully qualified non-R user)

**Pull request merged**